server/index.js

```diff
@@ -57,6 +60,106 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  if (!req.session.cartId) {
+    res.json([]);
+  } else {
+    const cartId = req.session.cartId;
+    const sql = `
+      select "ci"."cartItemId",
+             "ci"."price",
+             "p"."productId",
+             "p"."image",
+             "p"."name",
+             "p"."shortDescription"
+        from "cartItems" as "ci"
+        join "products" as "p" using ("productId")
+       where "ci"."cartId" = $1
+      `;
+    const params = [cartId];
+
+    db.query(sql, params)
+      .then(result => {
+        res.status(200).json(result.rows[0]);
+      })
+      .catch(err => console.error(err));
+  }
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = req.body.productId;
+  if (typeof productId === 'undefined') {
+    res.status(400).json({
+      error: 'Missing required value'
+    });
+  } else if (productId < 0 || isNaN(productId)) {
+    res.status(400).json({
+      error: `${productId} must be valid integer`
+    });
+  }
```

Comment:
I recommend using `return` statements instead of `else`. That way even when the `else if` is matched, it's not possible for the request handler to continue executing.